TESTS/mbedmicro-rtos-mbed/MemoryPool/main.cpp

```diff
@@ -0,0 +1,732 @@
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+#include "rtos.h"
+#include "MemoryPool.h"
+
+using namespace utest::v1;
+
+
+/* Enum used to select block allocation method. */
+typedef enum{
+    ALLOC,
+    CALLOC
+} AllocType;
+
+/* Structure for complex block type. */
+typedef struct{
+    int a;
+    char b;
+    int c;
+}COMPLEX_TYPE;
+
+/* Function to check if complex type object is cleared.*/
+bool comp_is_cleared(COMPLEX_TYPE *object)
+{
+    if(object->a == 0 &&
+       object->b == 0 &&
+       object->c == 0)
+    {
+        return true;
+    }
+
+    return false;
+}
+
+/* Function to check if complex type object holds specified values.*/
+bool comp_is_equal(COMPLEX_TYPE *object, int a, char b, int c)
+{
+    if(object->a == a &&
+       object->b == b &&
+       object->c == c)
+    {
+        return true;
+    }
+
+    return false;
+}
+
+/* Function to set complex type object fields.*/
+void comp_set(COMPLEX_TYPE *object, int a, char b, int c)
+{
+    object->a = a;
+    object->b = b;
+    object->c = c;
+}
+
+
+/* Template for functional tests for alloc(), calloc() functions
+ * of MemoryPool object.
+ *
+ * Given MemoryPool object of the specified type and queue size has
+ * been successfully created.
+ * When max number of blocks is allocated form the pool.
+ * Then all allocations are successful.
+ *
+ * */
+template<typename T, const uint32_t numOfEntries, AllocType atype>
+void test_mem_pool_alloc_success() {
+    MemoryPool <T, numOfEntries> *p_mem_pool;
+    T * p_blocks[numOfEntries];
+    uint32_t i;
+
+    /* Create a memory pool. */
```

Comment:
It is good to have comments but I find a lot of them redundant . like create a memory pool, it is obvious from the code.

the comment on line 80 below for instance is fine.